src/local-state/__tests__/LocalState/fixtures/local-resolvers-without-root.ts

```diff
@@ -0,0 +1,63 @@
+// Eventually this file will be auto generated by codegen after adding the codegen plugin
+import type { LocalState } from "@apollo/client/local-state";
+
+type Maybe<T> = T | null;
+
+type Resolver<
+  TResult,
+  TParent = Record<string, unknown>,
+  TArgs = Record<string, unknown>,
+> = LocalState.Resolver<TResult, TParent, TArgs>;
+
+export type Food = {
+  __typename: "Food";
+  categories?: Maybe<Array<FoodCategory>>;
+  name?: Maybe<string>;
+};
+
+export type FoodCategoriesArgs = {
+  limit?: Maybe<number>;
+  offset: number;
+};
+
+export enum FoodCategory {
+  Italian = "ITALIAN",
+}
+
+export type User = {
+  __typename: "User";
+  favoriteFood?: Maybe<Food>;
+  isLoggedIn?: boolean;
+};
+
+type ParentTypes = {
+  Food: Food;
+  Query: undefined;
+  User: User;
+};
```

Comment:
Just for visualisation:

I feel for me this would feel more natural instead of an arbitrary "root value"
```suggestion
export type Query = {
  __typename: "Query";
  currentUserId?: Maybe<string>;
};

type ParentTypes = {
  Food: Food;
  Query: Query;
  User: User;
};
```

Reply:
I'll revisit this once I tweak the `rootValue` option to be `context`. This is updated in https://github.com/apollographql/apollo-client/pull/12617 anyways and will be generated by codegen so we'll need to make this something that is statically analyzable. In https://github.com/apollographql/apollo-client/pull/12617 I added a `baseTypesPath` option that will import from a file that should contain all the schema files. I will need to tweak that as well to ensure the root query type is used as the parent type.

Reply:
I'm going to leave the types alone for now and get them fully updated in #12617. That `Query` parent type should get generated as `DeepPartial<QueryType>` in that PR once I get it updated.